### PR TITLE
[JIT] create jit module and implement rgb_to_grayscale 

### PR DESCRIPTION
**Pull request by edgarriba**

### CODE REVIEW

File: kornia/color/gray.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ class RgbToGrayscale(nn.Module):
     def __init__(self) -> None:
         super(RgbToGrayscale, self).__init__()
 
-    def forward(self, input: torch.Tensor) -> torch.Tensor:  # type: ignore
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
```
Comment (edgarriba) [inline kornia/color/gray.py]: @suo as we discussed, tests for jit seems to be working, however mypy triggers the following error:
```bash
Typechecker (mypy)
kornia/color/gray.py:30: error: Signature of "forward" incompatible with supertype "Module"
Makefile:21: recipe for target 'mypy' failed
make: *** [mypy] Error 1
```

any idea how should we fix that ? if I keep the `# type: ignore` then `torch.jit.script` seems to not do the job :D

Comment (suo) [reply kornia/color/gray.py]: Sorry for the delay. It looks like an interference between mypy's type system and TorchScript's. Can you file an issue to support `# type: ignore` on PyTorch? I think we can add it relatively easily.

Comment (edgarriba) [reply kornia/color/gray.py]: @suo done ;D https://github.com/pytorch/pytorch/issues/29099

Comment (suo) [reply kornia/color/gray.py]: Sorry for the late return on this, but https://github.com/pytorch/pytorch/pull/31057 should fix this specific issue.

Comment (edgarriba) [reply kornia/color/gray.py]: @suo thanks for the fix. We'll try it and integrate as soon the patch is included in the next stable release

Comment (edgarriba) [reply kornia/color/gray.py]: @@carlosb1 please, take a look at this

**Comment by ducha-aiki** [comment]
LGTM. I'll do features-jit next week.

**Pull request merged**